atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionStatusUtils.java

```diff
@@ -18,51 +18,61 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.palantir.atlasdb.transaction.service.TransactionStatus;
-import com.palantir.atlasdb.transaction.service.TransactionStatuses;
+import com.palantir.atlasdb.transaction.service.TransactionStatus.Aborted;
+import com.palantir.atlasdb.transaction.service.TransactionStatus.Committed;
 import com.palantir.logsafe.SafeArg;
 import com.palantir.logsafe.exceptions.SafeIllegalStateException;
 import java.util.Optional;
 import java.util.function.Function;
 
 public final class TransactionStatusUtils {
+
     private TransactionStatusUtils() {}
 
     public static TransactionStatus fromTimestamp(long timestamp) {
         if (timestamp == TransactionConstants.FAILED_COMMIT_TS) {
-            return TransactionConstants.ABORTED;
+            return TransactionStatus.aborted();
         }
-        return TransactionStatuses.committed(timestamp);
+        return TransactionStatus.committed(timestamp);
     }
 
     /**
      * This helper is only meant to be used for transactions with schema < 4. For schemas >= 4,
      * use {@link #getCommitTsFromStatus(long, TransactionStatus, Function)}
      * */
     public static long getCommitTimestampIfKnown(TransactionStatus status) {
-        return TransactionStatuses.caseOf(status)
-                .committed(Function.identity())
-                .aborted_(TransactionConstants.FAILED_COMMIT_TS)
-                .otherwise(() -> {
-                    throw new SafeIllegalStateException("Illegal transaction status", SafeArg.of("status", status));
-                });
+        if (status instanceof Committed) {
```

Comment:
In the future we can replace this with [sealed classes and pattern matching switch statements](https://openjdk.org/jeps/433).